zrml/prediction-markets/src/benchmarks.rs

```diff
@@ -230,29 +234,155 @@ fn setup_resolve_common_scalar_after_dispute<T: Config>(
     Ok((caller, market_id))
 }
 
+fn create_reported_market_with_pool<T: Config>(
+    categories: u32,
+) -> Result<(T::AccountId, MarketIdOf<T>), &'static str> {
+    let range_start: MomentOf<T> = 100_000u64.saturated_into();
+    let range_end: MomentOf<T> = 1_000_000u64.saturated_into();
+    let (caller, market_id) = create_market_common::<T>(
+        MarketCreation::Permissionless,
+        MarketType::Categorical(categories.saturated_into()),
+        ScoringRule::CPMM,
+        Some(MarketPeriod::Timestamp(range_start..range_end)),
+    )?;
+
+    let max_swap_fee: BalanceOf<T> = MaxSwapFee::get().saturated_into();
+    let min_liquidity: BalanceOf<T> = MinLiquidity::get().saturated_into();
+    let _ = Call::<T>::buy_complete_set { market_id, amount: min_liquidity }
+        .dispatch_bypass_filter(RawOrigin::Signed(caller.clone()).into())?;
+    let weight_len: usize = MaxRuntimeUsize::from(categories).into();
+    let weights = vec![MinWeight::get(); weight_len];
+    Pallet::<T>::deploy_swap_pool_for_market(
+        RawOrigin::Signed(caller.clone()).into(),
+        market_id,
+        max_swap_fee,
+        min_liquidity,
+        weights,
+    )
+    .unwrap();
+
+    let _ = Call::<T>::admin_move_market_to_closed { market_id }
+        .dispatch_bypass_filter(T::CloseOrigin::successful_origin())?;
+    let outcome = OutcomeReport::Categorical(0u16);
+    let _ = Call::<T>::report { market_id, outcome }
+        .dispatch_bypass_filter(RawOrigin::Signed(caller.clone()).into())?;
+
+    Ok((caller, market_id))
+}
+
 benchmarks! {
     admin_destroy_disputed_market{
-        let categories = T::MaxCategories::get();
-        let (caller, market_id) = setup_resolve_common_categorical::<T>(0, 0, categories)?;
+        let a in (T::MinCategories::get().into())..T::MaxCategories::get().into();
+        let d in 1..T::MaxDisputes::get();
+        let o in 0..63;
+        let c in 0..63;
+        let r in 0..63;
 
-        let categories : u32 = categories.saturated_into();
-        for i in 0..categories.min(T::MaxDisputes::get()) {
-            let origin = caller.clone();
-            let disputes = crate::Disputes::<T>::get(market_id);
-            let market = T::MarketCommons::market(&Default::default()).unwrap();
-            T::SimpleDisputes::on_dispute(&disputes, &market_id, &market)?;
+        let (caller, market_id) = create_reported_market_with_pool::<T>(a)?;
+
+        let pool_id = T::MarketCommons::market_pool(&market_id).unwrap();
+
+        // minimum of number of disputes and (categories - 1) because of `matches_outcome_report`
+        for i in 1..=(a - 1).min(d) {
+            let outcome = OutcomeReport::Categorical(i.saturated_into());
```

Comment:
Did you take the number of disputes into account with this suggestion? I mean it could be, that there is one dispute, but `a = 5` and then I would get the `matches_outcome_report` produced error.

Reply:
just a question, since d represents number of disputes I think shouldn't we use that for iteration? why number of categories should influence disputes reported to market?

Reply:
f922fcf9e8b779451c67333d24875f6a8a1fd632